File: abandon/addData.php
<?php
// 資料庫連線設定
$servername = "localhost";
$username = "test123";
$password = "test123"; // 預設 XAMPP 的 root 沒有密碼
$dbname = "aiotdb";

// 建立連線
$conn = new mysqli($servername, $username, $password, $dbname);

// 檢查連線
if ($conn->connect_error) {
    die("連線失敗: " . $conn->connect_error);
}

// 檢查是否有透過 GET 方法傳遞 temp 和 humid 參數
if (isset($_GET['temp']) && isset($_GET['humid'])) {

    // 取得參數並轉換為浮點數，以防 SQL 隱碼攻擊
    $temp = floatval($_GET['temp']);
    $humid = floatval($_GET['humid']);

    // 準備 SQL 寫入語法
    $sql = "INSERT INTO sensor (temp, humid) VALUES ($temp, $humid)";

    // 執行 SQL 語法並檢查是否成功
    if ($conn->query($sql) === TRUE) {
        echo "新資料寫入成功！溫度: $temp, 濕度: $humid";
    }
    else {
        echo "Error: " . $sql . "<br>" . $conn->error;
    }
}
else {
    echo "請提供 temp 及 humid 參數，例如：?temp=25.5&humid=60.0";
}

// 關閉連線
$conn->close();
?>

File: old/addData.py
from flask import Flask, jsonify
import mysql.connector
from mysql.connector import Error

app = Flask(__name__)

# 讀取資料庫連線設定
def get_db_connection():
    try:
        connection = mysql.connector.connect(
            host='localhost',
            database='aiotdb',
            user='test123',
            password='test123'
        )
        return connection
    except Error as e:
        print(f"Error connecting to MySQL: {e}")
        return None

# 新增資料 API
@app.route('/aiot/<temp>/<humid>', methods=['GET'])
def add_data(temp, humid):
    try:
        # 將字串轉換為浮點數
        temp_val = float(temp)
        humid_val = float(humid)
        
        conn = get_db_connection()
        if conn is None:
            return jsonify({"status": "error", "message": "Database connection failed"}), 500
            
        cursor = conn.cursor()
        
        # 準備 SQL 查詢，使用參數化查詢防止 SQL Injection
        sql = "INSERT INTO sensor (temp, humid) VALUES (%s, %s)"
        val = (temp_val, humid_val)
        
        cursor.execute(sql, val)
        conn.commit()
        
        return jsonify({
            "status": "success", 
            "message": f"Data inserted successfully. Temp: {temp_val}, Humid: {humid_val}"
        }), 200
        
    except ValueError:
        return jsonify({"status": "error", "message": "Invalid parameters. Temp and humid must be numbers."}), 400
    except Error as e:
        return jsonify({"status": "error", "message": f"Database error: {e}"}), 500
    finally:
        if 'conn' in locals() and conn.is_connected():
            cursor.close()
            conn.close()

if __name__ == '__main__':
    # 預設跑在 port 5000
    app.run(host='0.0.0.0', port=5000, debug=True)
